In [2]:
# rank 0
# Read the file and count the numbers after "Comm groups:"
with open('comm_output.txt', 'r') as file:
    content = file.read()

# Extract the part after "Comm groups:"
if "Comm groups:" in content:
    numbers_part = content.split("Comm groups:")[1]
    # Split the numbers and count them
    numbers = numbers_part.split()
    count = len(numbers)
    print(f"Count of numbers after 'Comm groups:': {count}")
    prev_num = numbers[0]
    transition = 0
    for n in numbers:
        if n != prev_num:
            transition += 1
            prev_num = n
    print(f"Number of transitions: {transition}")
else:
    print("No 'Comm groups:' found in the file.")

Count of numbers after 'Comm groups:': 884
Number of transitions: 117


## Reconfiguration backend
comm group(node_id)

rank 0: 3(8, TP)    ->1(7142, DP)                   ->3(6865, TP)                           ->1(80, DP, prev 31-trigger-success)    ->3(7140, TP)
rank 1: 4(8)        ->1(7142, prev 6854-trigger)    ->4(6865)                               ->1(80)                                 ->4(7140, prev 7139-trigger-adjacent)
rank 2: 3(8)        ->2(7142)                       ->3(6865, prev 7142-trigger-success)    ->2(80)                                 ->3(7140)
rank 3: 4(8)        ->2(7142)                       ->4(6865, prev )                        ->2(80)                                 ->4(7140)

## Analytical backend

rank 0: 3->1 ... (165 transitions)

## Comm group list
rank 0: 117 transitions



In [ ]:
Rank 0